File: Mix/xcpc/2024ccpc_harbin/C.py
t = int(input())
dic = {'N': 0, 'E': 1, 'S': 2, 'W': 3}
for _ in range(t):
    n = int(input())
    line = input().split()
    d1 = line[0]
    num = int(line[1])
    print(n * 2 - 1, end = ' ')
    print(d1)
    print('Z', end = ' ')
    print(num)
    for i in range(1, n):
        line = input().split()
        d2 = line[0]
        num = int(line[1])
        if (dic[d2] == (dic[d1] + 1) % 4):
            print('R')
        else:
            print('L')
        print('Z', end = ' ')
        print(num)
        d1 = d2

File: Mix/xcpc/2024ccpc_harbin/C.cpp
#include <bits/stdc++.h>

using namespace std;
int dic[256];

void solve(){
    int n; cin >> n;
    char d1; int num;
    cin >> d1 >> num;
    cout << n * 2 - 1 << ' ' << d1 << '\n';
    cout << 'Z' << ' ' << num << '\n';
    for (int i = 1; i < n; ++i) {
        char d2;
        cin >> d2 >> num;
        if (dic[d2] == (dic[d1] + 1) % 4)
            cout << 'R' << '\n';
        else
            cout << 'L' << '\n';
        cout << 'Z' << ' ' << num << '\n';
        d1 = d2;
    }
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    dic['N'] = 0;
    dic['E'] = 1;
    dic['S'] = 2;
    dic['W'] = 3;
    int t=1;
    cin>>t;
    while(t--){
        solve();
    }
    return 0;
}